infrastructure/src/test/java/org/corfudb/infrastructure/batchprocessor/BatchProcessorTest.java

```diff
@@ -0,0 +1,308 @@
+import org.corfudb.runtime.exceptions.WrongEpochException;
+import org.corfudb.runtime.proto.service.LogUnit;
+import org.corfudb.util.serializer.Serializers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static org.corfudb.protocols.CorfuProtocolCommon.DEFAULT_UUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolBase.getSealRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getLogAddressSpaceRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getRangeWriteLogRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getResetLogUnitRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getTailRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getTrimLogRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolLogUnit.getWriteLogRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getRequestMsg;
+import static org.corfudb.runtime.proto.service.CorfuMessage.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+public class BatchProcessorTest {
+
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+    private BatchProcessor batchProcessor;
+    private StreamLog mockStreamLog;
+    private final AtomicInteger requestCounter = new AtomicInteger();
+    private final long DEFAULT_SEAL_EPOCH = 1L;
+    private final long LARGER_SEAL_EPOCH = 5L;
+
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values. Note that the sealEpoch in BatchProcessor
+     * should equal to the epoch in request header.
+     *
+     * @return   the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, DEFAULT_SEAL_EPOCH,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values. Note that the sealEpoch in BatchProcessor
+     * should equal to the epoch in request header.
```

Comment:
nit: should be equal to the epoch in request header

Reply:
Added, thanks!